macros/base/homing/homing_overide.cfg

```diff
@@ -94,6 +93,13 @@ gcode:
             G0 Z{homing_zhop} F{z_drop_speed}
             {% set X, Y, Z = True, True, True %}
         {% endif %}
+    {% else %}
+        {% if verbose %}
+            { action_respond_info("Homing X or Y while Z is not homed, forcing ZHOP") }
+        {% endif %}
+        SET_KINEMATIC_POSITION Z=0
+        G0 Z{homing_zhop} F{z_drop_speed}
+        M84 Z
```

Comment:
I think it's not needed as we know for sure that Z is already not homed

Reply:
you described that setting the kinematic position would be like homing manually. this must be undone after the zhop